### PR TITLE
[RW-6450][risk=no] Use new api call for criteria menu

### PR DESCRIPTION
**Pull request by dolbeew**

### CODE REVIEW

**Comment by dolbeew** [comment]
Some Puppeteer tests will fail since the Surveys, Physical Measurements and Visits trees don't currently work with the new menu api call. I'll rerun once the menu table for test env has been updated.

File: ui/src/app/cohort-search/search-group-list/search-group-list.component.tsx [hunk 1]
```diff
@@ -167,38 +182,22 @@ const SearchGroupList = fp.flow(withCurrentWorkspace(), withCdrVersions())(
...
-          return acc;
-        }, {programTypes: [], domainTypes: []});
-        criteriaMenuOptions[cdrVersionId].programTypes.sort((a, b) => a.order - b.order);
-        criteriaMenuOptions[cdrVersionId].domainTypes.sort((a, b) => a.order - b.order);
+          return option;
+        }));
+        criteriaMenuOptions[cdrVersionId] = {
+          programTypes: menuOptions.filter(opt => opt.category === 'Program Data').sort(sortMenuItems),
+          domainTypes: menuOptions.filter(opt => opt.category === 'Domains').sort(sortMenuItems)
+        };
```
Comment (freemabd) [inline ui/src/app/cohort-search/search-group-list/search-group-list.component.tsx:200]: The API call does the sort logic in the database call. The list should always be sorted in the proper order in the response.

**Review by freemabd — COMMENTED** [review]
Just a comment about the API call doing the sort using the database.

File: ui/src/app/cohort-search/search-group-list/search-group-list.component.tsx (hunk 1 above)
Comment (dolbeew) [reply ui/src/app/cohort-search/search-group-list/search-group-list.component.tsx]: Ok, thanks. Removed the frontend sorting.

**Review by freemabd — APPROVED** [review]
Looks good.

**Pull request merged**